packages/schema/src/interfaces.ts

```diff
@@ -219,6 +219,29 @@ export interface IJGISFormSchemaRegistry {
   has(name: string): boolean;
 }
 
+export interface IControlPanelModel {
+  disconnect(f: any): void;
+  documentChanged: ISignal<IJupyterGISTracker, IJupyterGISWidget | null>;
+  filePath: string | undefined;
+  jGISModel: IJupyterGISModel | undefined;
+  sharedModel: IJupyterGISDoc | undefined;
+}
+
+/**
+ * Options of the left panel widget.
+ */
+export interface ILeftPanelOptions {
+  model: IControlPanelModel;
+  onSelect: ({ type, item, nodeId }: ILeftPanelClickHandlerParams) => void;
+}
+
+export interface ILeftPanelClickHandlerParams {
+  type: SelectionType;
+  item: string;
+  nodeId?: string;
+  event: MouseEvent;
+}
+
```

Comment:
These types are specific to the left panel, should they not stay under `src/panelview/`? They are not used outside of this directory as far as I can see

Reply:
I moved the `IControlPanelModel` back to the `types` in `base` and the other two into the `leftPanel` file.